### ISSUE TITLE
Tcl/Tk modules when building tkrplot in R 3.3.1

### ISSUE THREAD

**Issue by jhein32** [hpcugent/easybuild-easyconfigs#3418]
We have issues with R builds failing. It fails in the tkrplot extension. This concerns multiple easyconfigs e.g.:

-  R-3.2.3-foss-2016a.eb
-  R-3.3.1-foss-2016b.eb
-  R-3.3.1-intel-2016b.eb

Older configs not asking for tkrplot seem fine. Error message suggest an issue with tcl and/or tk to me. I checked that this is stated as a prerequisite in the config file and forced a rebuild of both packages. Below is a section from the log for R-3.3.1-foss-2016b.eb:

Thanks
Joachim

```
== 2016-08-16 13:55:03,375 build_log.py:152 ERROR cmd "R CMD INSTALL /home/swadmin/.local/easybuild/sources/r/R/extensions/tkrplot_0.0-23.tar.gz   --library=/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/library --no-clean-on-error" exited with exitcode 1 and output:
* installing *source* package ‘tkrplot’ ...
** package ‘tkrplot’ successfully unpacked and MD5 sums checked
checking for gcc... gcc
checking for C compiler default output... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables... 
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for X... libraries , headers 
checking for gethostbyname... yes
checking for connect... yes
checking for remove... yes
checking for shmat... yes
checking for IceConnectionNumber in -lICE... yes
checking for tclConfig.sh... no
checking for tclConfig.sh in library (sub)directories... no
checking for tkConfig.sh... no
checking for tkConfig.sh in library (sub)directories... no
checking for tcl.h... yes
checking for tk.h... yes
configure: creating ./config.status
config.status: creating src/Makevars
** libs
gcc -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/include -DNDEBUG  -I/sw/easybuild/software/Compiler/GCC/5.4.0-2.26/OpenBLAS/0.2.18-LAPACK-3.6.1/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/ScaLAPACK/2.0.2-OpenBLAS-0.2.18-LAPACK-3.6.1/include -I/sw/easybuild/software/MPI/GCC/4.9.3-2.25/OpenMPI/1.10.2/FFTW/3.3.4/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libreadline/6.3/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/ncurses/6.0/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/bzip2/1.0.6/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/XZ/5.2.2/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/PCRE/8.38/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libpng/1.6.23/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libjpeg-turbo/1.5.0/include -I/sw/easybuild/software/Core/Java/1.8.0_92/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/Tcl/8.6.5/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/Tk/8.6.5/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/cURL/7.49.1/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libxml2/2.9.4/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/GDAL/2.1.0/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/PROJ/4.9.2/include -I/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/GMP/6.1.1/include    -fpic  -O2 -march=native  -c tcltkimg.c -o tcltkimg.o
gcc -shared -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/lib -L/sw/easybuild/software/Core/GCCcore/5.4.0/lib64 -L/sw/easybuild/software/Core/GCCcore/5.4.0/lib -L/sw/easybuild/software/Compiler/GCC/5.4.0-2.26/OpenBLAS/0.2.18-LAPACK-3.6.1/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/ScaLAPACK/2.0.2-OpenBLAS-0.2.18-LAPACK-3.6.1/lib -L/sw/easybuild/software/MPI/GCC/4.9.3-2.25/OpenMPI/1.10.2/FFTW/3.3.4/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libreadline/6.3/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/ncurses/6.0/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/bzip2/1.0.6/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/XZ/5.2.2/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/PCRE/8.38/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libpng/1.6.23/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libjpeg-turbo/1.5.0/lib -L/sw/easybuild/software/Core/Java/1.8.0_92/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/Tcl/8.6.5/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/Tk/8.6.5/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/cURL/7.49.1/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/libxml2/2.9.4/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/GDAL/2.1.0/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/PROJ/4.9.2/lib -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/GMP/6.1.1/lib -o tkrplot.so tcltkimg.o -L/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/lib -lR
installing to /sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/library/tkrplot/libs
** R
** preparing package for lazy loading
Warning: S3 methods ‘as.character.tclObj’, ‘as.character.tclVar’, ‘as.double.tclObj’, ‘as.integer.tclObj’, ‘as.logical.tclObj’, ‘as.raw.tclObj’, ‘print.tclObj’, ‘[[.tclArray’, ‘[[<-.tclArray’, ‘$.tclArray’, ‘$<-.tclArray’, ‘names.tclArray’, ‘names<-.tclArray’, ‘length.tclArray’, ‘length<-.tclArray’, ‘tclObj.tclVar’, ‘tclObj<-.tclVar’, ‘tclvalue.default’, ‘tclvalue.tclObj’, ‘tclvalue.tclVar’, ‘tclvalue<-.default’, ‘tclvalue<-.tclVar’, ‘close.tkProgressBar’ were declared in NAMESPACE but not found
Error : .onLoad failed in loadNamespace() for 'tcltk', details:
  call: fun(libname, pkgname)
  error: Tcl/Tk support is not available on this system
Error : package ‘tcltk’ could not be loaded
ERROR: lazy loading failed for package ‘tkrplot’
 (at easybuild/software/Core/EasyBuild/2.8.2/lib/python2.7/site-packages/easybuild_framework-2.8.2-py2.7.egg/easybuild/tools/run.py:397 in parse_cmd_output)
== 2016-08-16 13:55:03,376 easyblock.py:2367 WARNING build failed (first 300 chars): cmd "R CMD INSTALL /home/swadmin/.local/easybuild/sources/r/R/extensions/tkrplot_0.0-23.tar.gz   --library=/sw/easybuild/software/MPI/GCC/5.4.0-2.26/OpenMPI/1.10.3/R/3.3.1/lib64/R/library --no-clean-on-error" exited with exitcode 1 and output:
* installing *source* package ‘tkrplot’ ...
** packa
== 2016-08-16 13:55:03,376 easyblock.py:270 INFO Closing log for application name R version 3.3.1
```

**Comment by verdurin** [comment]
@jhein32 - see https://github.com/hpcugent/easybuild-easyconfigs/issues/2749

**Comment by jhein32** [comment]
@verdurin Thanks.  The change described in #2749 fixes R-3.2.3-foss-2016a.eb.  I am still trying on the others.

@boegel If #2749 is enough to get the most recent R configs fixed, I am happy for this to be closed.

**Comment by boegel** [comment]
update to easyblock for R will take care of the missing options, cfr. https://github.com/hpcugent/easybuild-easyblocks/pull/995

let's follow up in #2749
